### PR TITLE
Add support for multiple variants per OS

### PR DESCRIPTION
**Pull request by jrstarke**
Also added a server variant of Debian for Rock 3a and Rock 5b

Build of this branch is up at https://github.com/jrstarke/debos-radxa/actions/runs/3554787054

### CODE REVIEW

**Comment by RadxaStephen** [comment]
Thanks. I will test it.

**Comment by jrstarke** [comment]
@RadxaStephen If you want to try a build, a few things to try:
- Do a build with the new configs in it. You should see a build for [rock-5b](https://github.com/jrstarke/debos-radxa/actions/runs/3554787054/jobs/5971095878) and [rock-3a debian server](https://github.com/jrstarke/debos-radxa/actions/runs/3554787054/jobs/5971095304).
- If you remove the config file, then those builds should go away. 
- Want a Debian Server or Ubuntu Desktop for a different board, just add the config

As for the builds themselves. I installed the rock-5b build on my rock-5b. I'm seeing [some errors in dmesg](https://gist.github.com/jrstarke/c77899f53418075121733ae385f1c7f9). Not sure if these are normal

File: configs/boards/rk3568/config-rock-3a-debian-bullseye-server-arm64-gpt.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
```
Comment (alexcb) [inline configs/boards/rk3568/config-rock-3a-debian-bullseye-server-arm64-gpt.sh:1]: nit: missing `set -e`

File: configs/boards/rk3568/config-rock-3a-debian-bullseye-server-arm64-gpt.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+{{- \$suite := or  .suite "${DISTRO}" -}}
+{{- \$imagesize := or .imagesize "${IMAGESIZE}" -}}
+{{- \$bootpartitionend := or .bootpartitionend "1081343S" -}}
+{{- \$rootpartitionstart := or .rootpartitionstart "1081344S" -}}
+{{- \$apt_repo := or .apt_repo "radxa" -}}
+
+EOF
+
+# Add images yaml
+cat > $BUILD_DIR/${BOARD}-${MODEL}-${DISTRO}-${VARIANT}-${ARCH}-${FORMAT}-yaml.list <<EOF
```
Comment (alexcb) [inline configs/boards/rk3568/config-rock-3a-debian-bullseye-server-arm64-gpt.sh:40]: nit: protect against vars having spaces.

```suggestion
cat > "$BUILD_DIR/${BOARD}-${MODEL}-${DISTRO}-${VARIANT}-${ARCH}-${FORMAT}-yaml.list" <<EOF
```

**Comment by jrstarke** [comment]
Updated build https://github.com/jrstarke/debos-radxa/actions/runs/3644630573

**Comment by RadxaYuntian** [comment]
This PR completely breaks [radxa/debos-build-action](https://github.com/radxa/debos-build-action), which does not provide `-v` option to `build-supported-configuration.sh`.

**Comment by RadxaYuntian** [comment]
Even after adding `-v` option in [radxa/debos-build-action](https://github.com/radxa/debos-build-action) it still regressed on previously possible configuration: [old](https://github.com/radxa-build/radxa-nx5-io/actions/runs/3814537231/jobs/6488917488#step:4:517) vs [new](https://github.com/radxa-build/radxa-nx5-io/actions/runs/3922365629/jobs/6705260125#step:4:495).

**Comment by RadxaYuntian** [comment]
Fixed typo and now it seems working.

**Pull request merged**